File: rotatedDigits.py
class Solution:
    def isGood(self,val):
        flag = 0
        
        while(val != 0):
            if(val%10 == 2 or val%10 == 5 or val%10 == 6 or val%10 == 9):
                flag = 1
            if(val%10 == 3 or val%10 == 4 or val%10 == 7):
                flag = 0
                break
            val = val//10
        
        if(flag == 1):
            return 1
        
        return 0
            
    def rotatedDigits(self, n: int) -> int:
        ans = 0
        
        for i in range(1,n+1):
            if(self.isGood(i)):
                ans += 1
        
        return ans

File: rotatedDigits.java
class Solution {
    public int rotatedDigits(int n) {
        int[] dp = new int[n+1];
        int ans = 0;
        
        for(int i=0;i<n+1;i++)
        {
            if(i<10)
            {
                if(i == 0 || i == 1 || i == 8)
                {
                    dp[i] = 1;
                }
                else if(i == 2 || i == 5 || i == 6 || i == 9)
                {
                    dp[i] = 2;
                    ans++;
                }
            }
            else
            {
                int a = dp[i%10], b = dp[i/10];
                if(a == 0 || b == 0)
                {
                    dp[i] = 0;
                }
                else if(a == 1 && b == 1)
                {
                    dp[i] = 1;
                }
                else
                {
                    dp[i] = 2;
                    ans++;
                }
            }
        }
        
        return ans;
    }
}
